test/integration/upload-notebook-test.js

```diff
@@ -0,0 +1,171 @@
+/**
+ * Copyright (c) Jupyter Development Team.
+ * Distributed under the terms of the Modified BSD License.
+ */
+
+var sinon = require('sinon');
+var chai = require('chai').use(require('sinon-chai'));
+var expect = chai.expect;
+var request = require('request');
+var fs = require('fs');
+var path = require('path');
+var urljoin = require('url-join');
+var rimraf = require('rimraf');
+
+var appUrl = process.env.APP_URL;
+var uploadUrl = urljoin(appUrl, '/notebooks/uploadtest');
+var dataDir = './data';
+var dataDir2 = './data2';
+
+describe('upload notebook', function() {
+    before(function(done) {
+        // move existing data to preserve it
+        fs.rename(dataDir, dataDir2, function(err) {
+            if (err) { console.error(err); }
+            fs.mkdir(dataDir, function(err) {
+                if (err) { console.error(err); }
+                done();
+            });
+        });
+    });
+
+    after(function(done) {
+        // delete test data and move original data back
+        rimraf(dataDir, function(err) {
+            if (err) { console.error(err); }
+            fs.rename(dataDir2, dataDir, function(err) {
+                if (err) { console.error(err); }
+                done();
+            });
+        });
+    });
```

Comment:
Couple of things:

1. We run integration tests in its own Docker container, separate from that of the proxy server. So I don't think moving this folder is necessary. @parente, am I understanding Docker functionality correctly?
2. I would prefer to treat the server as a black box during the integration tests. In this case, I would just upload your test files to a unique name (maybe generate a UUID-type string for the name).